src/wp-admin/includes/class-wp-filesystem-direct.php

```diff
@@ -331,11 +331,37 @@ public function move( $source, $destination, $overwrite = false ) {
 			return false;
 		}
 
+		if ( $overwrite && $this->exists( $destination ) && ! $this->delete( $destination, true ) ) {
+			// Can't overwrite if the destination couldn't be deleted.
+			return false;
+		}
+
 		// Try using rename first. if that fails (for example, source is read only) try copy.
 		if ( @rename( $source, $destination ) ) {
+			if ( $this->is_dir( $destination ) ) {
+				/*
+				* When using an environment with shared folders,
+				* there is a delay in updating the filesystem's cache.
+				*
+				* This is a known issue in environments with a VirtualBox provider.
+				*
+				* A 200ms delay gives time for the filesystem to update its cache,
+				* prevents "Operation not permitted", and "No such file or directory" warnings.
+				*
+				* This delay is used in other projects, including Composer.
+				* @link https://github.com/composer/composer/blob/main/src/Composer/Util/Platform.php#L228-L233
+				*/
+				usleep( 200000 );
+				wp_opcache_invalidate_directory( $destination );
```

Comment:
Should we have these here, or only in `move_dir()`, or just leave it up to extenders to determine when it's needed for their implementations? While this is _still_ faster than the current option of `copy_dir()` + `::delete()`, it may not be preferable to make this decision for extenders.

IMHO, as it's faster than any alternative in Core, it's worth having the `usleep( 200000 )` here to avoid destructive results on VirtualBox, and `wp_opcache_invalidate_directory()` for good practice, which will benefit users whether extenders call `::move()` or `move_dir()` in their plugins/themes.